runtime/interpreter/value.go

```diff
@@ -15040,6 +15116,104 @@ func NewEnumCaseValue(
 	return v
 }
 
+func (v *CompositeValue) getBaseValue(interpreter *Interpreter, locationRange LocationRange) *EphemeralReferenceValue {
+	return v.base
+}
+
+func (v *CompositeValue) setBaseValue(interpreter *Interpreter, base *CompositeValue) {
+	attachmentType, ok := interpreter.MustSemaTypeOfValue(v).(*sema.CompositeType)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+
+	var baseType sema.Type
+	switch ty := attachmentType.GetBaseType().(type) {
+	case *sema.InterfaceType:
+		baseType, _ = ty.RewriteWithRestrictedTypes()
+	default:
+		baseType = ty
+	}
+
+	// the base reference can only be borrowed with the declared type of the attachment's base
+	v.base = NewEphemeralReferenceValue(interpreter, false, base, baseType)
+}
+
+func attachmentMemberName(ty sema.Type) string {
+	return attachmentNamePrefix + string(ty.ID())
+}
+
+func (v *CompositeValue) getAttachmentValue(interpreter *Interpreter, locationRange LocationRange, ty sema.Type) *CompositeValue {
+	if attachment := v.GetMember(interpreter, locationRange, attachmentMemberName(ty)); attachment != nil {
+		return attachment.(*CompositeValue)
+	}
+	return nil
+}
+
+func (v *CompositeValue) forEachAttachment(interpreter *Interpreter, locationRange LocationRange, f func(*CompositeValue)) {
```

Comment:
Maybe remove iteration-related code from PR for now

Reply:
As mentioned here (https://github.com/onflow/cadence/pull/2113#discussion_r1038252290), it's not unused